src/traces/waterfall/attributes.js

```diff
@@ -76,6 +77,15 @@ module.exports = {
     hovertext: barAttrs.hovertext,
     hovertemplate: barAttrs.hovertemplate,
 
+    // labels (legend is handled by plots.attributes.showlegend and layout.hiddenlabels)
+    textinfo: extendFlat({}, pieAtts.textinfo, {
+        editType: 'plot',
+        flags: ['text', 'delta', 'initial', 'final'],
```

Comment:
I think these flags are :ok_hand: 

We could maybe add `label`, but it might awkward as for `waterfall` it doesn't match a `labels` data array like for pie and sunburst traces. Maybe adding plain `'x'` and `'y'` flags would be best.